### PR TITLE
ws: Add --for-tls-proxy option

### PR DESCRIPTION
**Pull request by martinpitt**
If cockpit-ws runs behind a reverse proxy that does the TLS termination,
it rejects non-GnuTLS connections with a https:// origin with

    received request from bad Origin: https://localhost:9090

This previously required setting `Origins` in cockpit.conf to make this
work. Introduce a new `--for-tls-proxy` option to cockpit-ws which
switches the default Origin checking to https-only. With that, no
cockpit.conf modifications are required for local proxies.

This is implemented by a new `for-tls-proxy` property in
`CockpitWebServer`, which gets plumbed through the Origins check in
`cockpit_web_service_create_socket()`.

This is mostly intended for the upcoming external "cockpit-tls" proxy
which externalizes TLS termination, so that this does not have to modify
cockpit.conf or inspect/modify the HTTP stream. But it's generally
applicable to any local reverse TLS proxy.

Fix test-handlers to not call cockpit_handler_socket() with a NULL
server, so that reading the property does not cause a warning.

TODO:
 - [x] WebServer unit test (?)
 - [x] manpage update
 - [x] add socat to debian/ubuntu images for the new test: PR #11815 
 - [x] add socat to rhel images for the new test: PR #11816

### CODE REVIEW

**Comment by martinpitt** [comment]
First pass to get test results on all OSes. I can add socat to the rhel and debian images, so that we can enable the test there as well, but that doesn't need to be a blocker.

**Comment by martinpitt** [comment]
I added a unit test for the WebServer part as well now. The two unit tests plus the integration tests should be good enough IMHO, but I'll leave that verdict to the reviewer. :-)

**Comment by martinpitt** [comment]
Ah, for Debian/Ubuntu the cockpit-ws path is different, need to adjust the tests.

**Comment by martinpitt** [comment]
Fixed the tests on Debian/Ubuntu.

File: src/ws/cockpitwebservice.c [hunk 1]
```diff
@@ -1360,6 +1363,7 @@ cockpit_web_service_create_socket (const gchar **protocols,
  * @input_buffer: optional bytes already parsed after headers
  * @auth: authentication object
  * @creds: credentials of user or NULL for failed auth
+ * @for_tls_proxy: Accept (only) https:// origins on http connection by default
```
Comment (mvollmer) [inline src/ws/cockpitwebservice.c]: What about "Assume that the Browser is making TLS connections that are terminated in a proxy."   Changing the default Origin is the effect, but there might be more.

**Comment by martinpitt** [comment]
@mvollmer: Fixed, thanks!

**Pull request merged**